sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ComplexTypeSuite.scala

```diff
@@ -460,24 +460,38 @@ class ComplexTypeSuite extends SparkFunSuite with ExpressionEvalHelper {
   }
 
   test("error message of ExtractValue") {
-    val structType = StructType(StructField("a", StringType, true) :: Nil)
-    val otherType = StringType
-
-    def checkErrorMessage(
-      childDataType: DataType,
-      fieldDataType: DataType,
-      errorMessage: String): Unit = {
-      val e = intercept[org.apache.spark.sql.AnalysisException] {
+    checkError(
+      exception = intercept[AnalysisException](
         ExtractValue(
```

Comment:
Is it possible to reproduce the issue from user space? Using a SQL query, for instance.

Reply:
Yeah, if it's not an user-facing we should turn this into `INTERNAL_ERROR`

Reply:
@MaxGekk I've added 3 test cases in `org.apache.spark.sql.errors.QueryExecutionErrorsSuite` please check

Reply:
Please, remove duplicate tests here. The added tests are enough.

Reply:
Done.